addons/stock/models/stock_picking.py

```diff
@@ -513,7 +513,10 @@ def write(self, vals):
             after_vals['location_dest_id'] = vals['location_dest_id']
         if after_vals:
             self.mapped('move_lines').filtered(lambda move: not move.scrapped).write(after_vals)
-        if vals.get('move_lines'):
+        # Do not run autoconfirm if any of the moves has an initial demand. If an initial demand
+        # is present in any of the moves, it means the picking was created through the "planned
+        # transfer" mechanism.
+        if vals.get('move_lines') and (self.state != 'draft' or not any(self.mapped('move_lines.product_uom_qty'))):
```

Comment:
and not (self.state == 'draft' and any(self.mapped('move_lines.product_uom_qty')))